docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+
+### Allow other boxes to talk to the secure box
+
+Coming soon.
+
+## Wrap-up
+[Go to top](#overview)
+
+In this guide we showed you how to:
+
+* Enable uVisor on an existing application.
+* Add a secure box to your application.
+    * Protect static and dynamic memories in a secure box.
+    * Gain exclusive access to a peripheral and an IRQ in a secure box.
+    * (Coming soon) Expose public secure entry points to a secure box.
+
+You can now modify the example or create a new one to protect your resources into a secure box. You might find the following resources useful:
+
+* [The uVisor API documentation](APIS.md)
+* [Debugging uVisor](DEBUGGING.md)
+
+If you found any bug or inconsistency in this guide, please [raise an issue](https://github.com/ARMmbed/uvisor/issues/new).
+
+## Appendix
+[Go to top](#overview)
+
+This section contains additional information that you might find useful when setting up a secure box.
+
+### The NVIC APIs
+
+The ARM CMSIS header files provide APIs to configure, enable and disable IRQs in the NVIC module. These APIs are all prefixed with `NVIC_` and can be found in the `core_cm*.h` files in your CMSIS module.
+
+In addition, the CMSIS header also provide APIs to set and get an interrupt vector at runtime. This require the vector table, which is usually located in flash, to be relocated to SRAM.
+
+When the uVisor is enabled, all NVIC APIs are re-routed to the corresponding uVisor vIRQ APIs, which virtualise the interrupt module. The uVisor interrupts model has the following features:
+
+* The uVisor owns the vector table.
+* All ISRs are relocated to SRAM.
+* Code in a box can only change the state of an IRQ (enable it, change its priority, etc.) if the box registered that IRQ with uVisor at runtime, using the `NVIC_SetVector` API.
+* An IRQ that belongs to a box can only be modified when that box context is active.
+
+Although this behaviour is different from the original NVIC one, it is backwards compatible. This means that legacy code (like a device HAL) will still work after uVisor is enabled. The general use case is the following:
+
+```C
+#define MY_IRQ 42
+
+/* Set the ISR for MY_IRQ at runtime.
+ * Without uVisor: Relocate the vector table to SRAM and set my_isr as the ISR for MY_IRQ.
+ * With    uVisor: Register MY_IRQ for the current box with my_isr as ISR. */
+NVIC_SetVector(MY_IRQ, &my_isr);
+
+/* Change the IRQ state. */
+NVIC_SetPriority(MY_IRQ, 3);
+NVIC_EnableIRQ(MY_IRQ);
+```
+
+> **Note**: In this model a call to `NVIC_SetVector` must always happen before an IRQ state is changed. In platforms that don't relocate the vector table such call might be originally absent and must be added to work with uVisor.
+
+For more information on the uVisor APIs, checkout the [uVisor APIs documentation](APIS.md).
```

Comment:
This appears to be a dead link. Consider omitting this link until the link is no longer dead.